pom.xml

```diff
@@ -6,14 +6,14 @@
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-spanner-hibernate</artifactId>
   <packaging>pom</packaging>
-  <version>3.9.7-SNAPSHOT</version><!-- {x-version-update:google-cloud-spanner-hibernate:current} -->
+  <version>4.0.0-SNAPSHOT</version><!-- {x-version-update:google-cloud-spanner-hibernate:current} -->
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>11</maven.compiler.source>
-    <maven.compiler.target>11</maven.compiler.target>
-    <java.version>11</java.version>
-    <hibernate.version>6.6.27.Final</hibernate.version>
+    <maven.compiler.source>17</maven.compiler.source>
```

Comment:
We should make sure to document this in the release notes once we get to that point (min required Java version 11 -> 17)